src/Points/DeintersectorOrderer.cc

```diff
@@ -43,44 +43,44 @@ bool DeintersectorPointsOrderer::isIntersecting(sf::Vector2f a, sf::Vector2f b,
 }
 
 
-void DeintersectorPointsOrderer::_checkIntersection(size_t idxA, size_t idxB, size_t idxC, size_t idxD)
+bool DeintersectorPointsOrderer::_handleIntersection(ActiveIterT seg1, ActiveIterT seg2)
 {
   auto& pts = *_pts;
-  if(isIntersecting(pts[idxA], pts[idxB], pts[idxC], pts[idxD])) {
-    _addIntersection(std::min(idxA, idxB), std::min(idxC, idxD));
-    auto it = _active.find({idxC, idxD});
-    if (it == _active.end())
-      throw std::runtime_error("It ain't supposed to happen, ever!");
-    _active.erase(it);
+  bool intersect = isIntersecting(pts[seg1->first], pts[seg1->second],
+                                  pts[seg2->first], pts[seg2->second]);
+  if(intersect) {
+    _addIntersection(std::min(seg1->first, seg1->second),
+                     std::min(seg2->first, seg2->second));
+    _active.erase(seg1);
+    _active.erase(seg2);
   }
+  return intersect;
 }
 
 void DeintersectorPointsOrderer::_handleStartPoint(size_t idxA, size_t idxB)
 {
+  _current_x = _pts->at(idxA).x;
   auto it = _active.insert({idxA, idxB}).first;
-  if (it != _active.begin()) {
+  if(it != _active.begin()) {
     auto dw = std::prev(it);
-    _checkIntersection(idxA, idxB, dw->first, dw->second);
+    if(_handleIntersection(it, dw))
+      return;
   }
   auto up = std::next(it);
```

Comment:
When we remove a segment, we still have to check whether its neighbors intersect (like in _handleEndPoint function). It's not strictly necessary, but it will find more intersections.